src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/missingnullcaseinswitch/InputMissingNullCaseInSwitchWithStringLiterals.java

```diff
@@ -0,0 +1,119 @@
+/*
+MissingNullCaseInSwitch
+
+*/
+
+//non-compiled with javac: Compilable with Java17
+package com.puppycrawl.tools.checkstyle.checks.coding.missingnullcaseinswitch;
+
+public class InputMissingNullCaseInSwitchWithStringLiterals {
+
+    void testSwitchRule(String obj) {
+        switch (obj) {   // violation, 'Switch using reference types should have a null case.'
+            default -> {}
```

Comment:
Is null case is a feature of jdk17 or 21 ?
If of 21, we should not do violations on jdk17 code.
If jdk17, we need to update documentation description of Check

Reply:
17 as a preview 21 final. That's why I put 21 in the doc because it is the final. Do you think we should put the first preview?

Reply:
If 21 is official, we should have some way to not violate jdk17 code that doesn't use preview features in compiler.

https://www.google.com/amp/s/www.geeksforgeeks.org/string-in-switch-case-in-java/amp/

We can not do violations on jdk17 code.

@nrmancuso , do we need jdk property to Check ?

Nuance is that in big companies checkstyle configuration is single file and projects are on different jdk due to long migration. Some old projects stay on old jdk for very long.

Reply:
I don't think we need property here. This is a single decision either you have a new enough Java version then you can use the check or an old java version so you will not use it.

> Nuance is that in big companies checkstyle configuration is single file and projects are on different jdk  

Then don't use this check at all. This check is applicable only for 17-18-19-20 previews and Java 21+

Am I getting you correctly? can you share a specific example where we need a property in the check

Reply:
Usually all Checks that we have can run on any jdk. 
At least we need to clearly state this in description that usage on jdk before 21 is prone to false positives and advised to not be used.

So user will do decision himself on how to  activate this Check.

Reply:
> At least we need to clearly state this in description that usage on jdk before 21 is prone to false positives and advised to not be used.

Yes, we can provide some details about Java version in the description, this is a good idea. As far as false positives go, we don’t need to mention this. We can just talk about when `when` was released and how this version of Java is the intended target of this check. 

We do not need a property here. You either use the check or not.

Reply:
> As of Java 21, it is now possible to integrate a null check within the switch, eliminating the risk of NullPointerException

Is this enough?